### ISSUE TITLE
Return empty block with column names/types over HTTP using Native Format if query returns no results

### ISSUE THREAD

**Issue by genzgd** [ClickHouse/ClickHouse#55939]
For clients that use Native format over HTTP (`clickhouse-connect`, `clickhouse-go`), a query that returns no results returns no data at all.  In many cases it would be useful for the application code to be able to construct an empty result with the proper column names and types, but this is not possible since that information isn't returned.

It would be useful and simplify the client code in these situations if ClickHouse returned an empty block containing column names and types instead of a completely empty response.

**Comment by wizzard0** [comment]
Yeah, this is important when eg showing UI, checking if we need to apply table migrations, etc
